reader.go

```diff
@@ -824,3 +1819,25 @@ func (r *reader) withErrorLogger(do func(*log.Logger)) {
 		r.withLogger(do)
 	}
 }
+
+// extractTopics returns the unique list of topics represented by the set of
+// provided members
+func extractTopics(members []memberGroupMetadata) []string {
+	visited := map[string]struct{}{}
```

Comment:
Style: keeping one variable declaration style within the local scope helps with readability, for example here you could rewrite this as
```go
var visited = map[string]struct{}{}
var topics []string
```